### PR TITLE
#11573 Remove Trac documentation build tools

### LINKED ISSUE #11573: Cleanup docs generation after full Read The Docs migration
We now only publish the documentation via the Read the docs project.

There are still some scripts and tools from the time when they were hosted on a dornkirk

While working on this, also move the dev docs into the root and generate redirect pages.

As part of this we can remove custom them and only use the read the docs theme

Remove trac-links and other plugins

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope and purpose

Fixes #11573

We no longer use twisted-web to host the doccs or Trac theme to integrate the docs to trac theme.
We also no longer build the documentation to be distributed on a separate site.
We only have Read The Docs.

This is cleanup for the code that is no longer used.

The docs are now hosted on Read The docs with build defined in `.readthedocs.yml` and 'docs/conf.py`

The docs are generated for dev using tox via `tox -e narrativedocs` that calls sphinx-build directly.

I think that by removing the script, it should be easier to understand how the docs are generated.

We no longer have some automated tests, but at the same time, we no longer have custom Sphinx theme or custom extensions that we need to mantain.

## How to test

* Check that you can build the docs locally via `tox -e narrativedocs`
* Check the Read The Docs version at https://twisted--11576.org.readthedocs.build/en/11576/

## Contributor Checklist:

* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://docs.twistedmatrix.com/en/stable/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://github.com/twisted/trac-wiki-archive/blob/trunk//ReviewProcess.mediawiki#News_files))
* [x] The title of the PR starts with the associated issue number (without the `#` character).
* [NA] I have updated the automated tests and checked that all checks for the PR are green.
* [x] The merge commit will use the below format
  The first line is automatically generated by GitHub based on PR ID and branch name.
  The other lines generated by GitHub should be replaced.

```
Merge pull request #123 from twisted/4356-branch-name-with-issue-id

Author: adiroiban
Fixes #11573

Remove tools used to use documentation for Trac integration.
```

### CODE REVIEW

**Comment by adiroiban** [comment]
I think that this is ready for review 

needs-review

**Comment by adiroiban** [comment]
just test please review

**Comment by adiroiban** [comment]
testing permission level
 please review

**Comment by adiroiban** [comment]
testing the review bot
needs-review

**Comment by danuker** [comment]
The bot was deactivated; sorry. needs-review

**Comment by adiroiban** [comment]
let's test
please review

**Comment by adiroiban** [comment]
ok... so this is ready for review.

please review

**Comment by danuker** [comment]
testing

please review

**Comment by danuker** [comment]
testing again
please review

**Comment by danuker** [comment]
please review
should work now

**Comment by adiroiban** [comment]
this is ready
please review

File: docs/conf.py [hunk 1]
```diff
@@ -194,18 +185,6 @@
...
-if on_rtd:
-    # For a PR the link is like:
-    # https://twisted--1422.org.readthedocs.build/en/1422/
-    # For a release:
-    # https://docs.twistedmatrix.com/en/twisted-20.3.0/
-    # https://docs.twistedmatrix.com/en/latest/
-    api_base_url = "/{}/{}/api/".format(
-        os.environ["READTHEDOCS_LANGUAGE"],
-        os.environ["READTHEDOCS_VERSION"],
-    )
```
Comment (exarkun) [inline docs/conf.py:208]: On RTD, we used to set this `api_base_url`.  Now we're always on RTD and we don't set it?  What will  `api_base_url` be?

File: src/twisted/python/_release.py [hunk 2]
```diff
@@ -20,22 +20,9 @@
...
-    "https://cryptography.io/en/latest/objects.inv",
-    "https://pyopenssl.readthedocs.io/en/stable/objects.inv",
-    "https://hyperlink.readthedocs.io/en/stable/objects.inv",
-    "https://twisted.org/constantly/docs/objects.inv",
-    "https://twisted.org/incremental/docs/objects.inv",
-    "https://python-hyper.org/projects/hyper-h2/en/stable/objects.inv",
-    "https://priority.readthedocs.io/en/stable/objects.inv",
-    "https://zopeinterface.readthedocs.io/en/latest/objects.inv",
-    "https://automat.readthedocs.io/en/latest/objects.inv",
-]
```
Comment (exarkun) [inline src/twisted/python/_release.py:38]: Do we get to drop these because of the upgraded Sphinx dependency?

**Review by exarkun — APPROVED** [review]
Thanks.  This looks like a nice collection of baggage to jettison.  I left a couple minor comments inline but they really just demonstrate my incomplete understanding of how RTD works.  And they're on deleted code so I don't think there's even any comments to add to the implementation.

Looking at the PR build on RTD it seems to be working fine - so please go ahead and merge.

File: docs/conf.py (hunk 1 above)
Comment (adiroiban) [reply docs/conf.py:208]: api_base_url - was used as a hack to generate links from Sphinx to pydoctor.

But now pydoctor generated an intersphinx object inventory, an Sphinx can use that for the links.

So the `api_base_url` is never used and should never be used.

Its usage was removed a long time ago, but we forgot to clean this part.

Comment (adiroiban) [reply docs/conf.py:208]: Ah... and we now run the same build process locally and on  read the docs

To keep it simple.

---------

In the past, we used to have a custom Sphinx theme, with the goal of integrating with the old website.

But we no longer have an old website ... and the docs are only hosted on Read The docs, with the default RTD theme.

------

File: src/twisted/python/_release.py (hunk 2 above)
Comment (adiroiban) [reply src/twisted/python/_release.py:38]: Good question.

This is a drive-by fix.

This list was used by pydoctor for API docs.
It was moved to setup.cfg in an older PR ...but I forgot to also delete this.

**Comment by adiroiban** [comment]
many thanks for the review.

I have enabled auto-merge. I hope it will be green soon.

**Pull request merged**